### PR TITLE
STG75 Python APIView

### PR DESCRIPTION
**Pull request by tasherif-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py [hunk 1]
```diff
@@ -406,6 +406,160 @@ def _upload_blob_options(  # pylint:disable=too-many-statements
...
+        if content_settings:
+            kwargs['blob_headers'] = BlobHTTPHeaders(
+                blob_cache_control=content_settings.cache_control,
+                blob_content_type=content_settings.content_type,
+                blob_content_md5=bytearray(content_settings.content_md5) if content_settings.content_md5 else None,
+                blob_content_encoding=content_settings.content_encoding,
+                blob_content_language=content_settings.content_language,
+                blob_content_disposition=content_settings.content_disposition
+            )
+        cpk = kwargs.pop('cpk', None)
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py:434]: We never used it right

File: sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py [hunk 2]
```diff
@@ -406,6 +406,160 @@ def _upload_blob_options(  # pylint:disable=too-many-statements
...
+            If the source is in another account, the source must either be public
+            or must be authenticated via a shared access signature. If the source
+            is public, no authentication is required.
+            Examples:
+            https://myaccount.blob.core.windows.net/mycontainer/myblob
+
+            https://myaccount.blob.core.windows.net/mycontainer/myblob?snapshot=<DateTime>
+
+            https://otheraccount.blob.core.windows.net/mycontainer/myblob?sastoken
+        :param bool copy_source_blob_properties:
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py:483]: Does other platforms also call this copy_*? Feel it’s like for a copy operation

File: sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py [hunk 3]
```diff
@@ -406,6 +406,160 @@ def _upload_blob_options(  # pylint:disable=too-many-statements
...
+        :keyword source_lease:
+            Specify this to perform the Copy Blob operation only if
+            the lease ID given matches the active lease ID of the source blob.
+        :paramtype source_lease: ~azure.storage.blob.BlobLeaseClient or str
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :keyword ~azure.storage.blob.ContentSettings content_settings:
+            ContentSettings object used to set blob properties. Used to set content type, encoding,
+            language, disposition, md5, and cache control.
+        :keyword ~azure.storage.blob.CustomerProvidedEncryptionKey cpk:
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py:540]: We are not using it

File: sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py [hunk 4]
```diff
@@ -406,6 +406,160 @@ def _upload_blob_options(  # pylint:disable=too-many-statements
...
+            Encrypts the data on the service-side with the given key.
+            Use of customer-provided keys must be done over HTTPS.
+            As the encryption key itself is provided in the request,
+            a secure connection must be established to transfer the key.
+        :keyword str encryption_scope:
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py:545]: We are not using this

Comment (xiafu-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py:545]: This comment is based on _upload_blob_from_url_options

Probably I should refer to the spec

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_share_client_async.py [hunk 5]
```diff
@@ -378,6 +389,9 @@ async def set_share_properties(self, **kwargs):
             Must be greater than 0, and less than or equal to 5TB.
         :keyword int timeout:
             The timeout parameter is expressed in seconds.
+        :keyword root_squash: Root squash to set on the share.  Only valid for
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_share_client_async.py:392]: format

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_share_client_async.py [hunk 6]
```diff
@@ -378,6 +389,9 @@ async def set_share_properties(self, **kwargs):
...
+         NFS shares. Possible values include: 'NoRootSquash', 'RootSquash', 'AllSquash'
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_share_client_async.py:393]: indentation

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_share_client_async.py [hunk 7]
```diff
@@ -393,13 +407,15 @@ async def set_share_properties(self, **kwargs):
         timeout = kwargs.pop('timeout', None)
         access_tier = kwargs.pop('access_tier', None)
         quota = kwargs.pop('quota', None)
+        root_squash = kwargs.pop('root_squash', None)
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_share_client_async.py:410]: you don't have to pop, root_squash will be passed into set_properties() through kwargs on line 420

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_share_client_async.py [hunk 8]
```diff
@@ -185,6 +185,11 @@ async def create_share(self, **kwargs):
 
         :keyword int timeout:
             The timeout parameter is expressed in seconds.
+        :keyword enabled_protocols: Protocols to enable on the share.
+        :paramtype enabled_protocols: str or ~azure.storage.fileshare.EnabledProtocols
+        :keyword root_squash: Root squash to set on the share.  Only valid for
+         NFS shares. Possible values include: 'NoRootSquash', 'RootSquash', 'AllSquash'.
+        :paramtype root_squash: str or ~azure.storage.fileshare.ShareRootSquash
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_share_client_async.py:192]: align the format with `timeout`

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py [hunk 9]
```diff
@@ -314,6 +314,11 @@ def create_share(self, **kwargs):
 
         :keyword int timeout:
             The timeout parameter is expressed in seconds.
+        :keyword enabled_protocols: Protocols to enable on the share.
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py:317]: enabled_protocols should be a list, else it's enabled_protocol.
If I remember properly, Sean said currently only a single protocol, while later we will support bunch

Comment (xiafu-msft) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py:317]: https://github.com/Azure/azure-sdk-for-java/pull/17239/files#diff-fb6910d874ec538dd2a452d3aaca8b72cc17c8f9fb70b23f4a7fd020bb6c632eR57-R68

File: sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py (hunk 2 above)
Comment (tasherif-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py:483]: This is not available in other apis. This is only specific to the new API. This is one of the parameters the generated code accepts https://github.com/Azure/azure-sdk-for-python/blob/46362a0d75d088c647fa2131d2ef3062db9214b7/sdk/storage/azure-storage-blob/azure/storage/blob/_generated/aio/operations_async/_block_blob_operations_async.py#L242

**Pull request closed without merging**